src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Legacy/RoslynCSharpTokenizer.cs

```diff
@@ -224,56 +128,26 @@ protected override string GetTokenContent(SyntaxKind type)
                         return "\t";
                     }
                     break;
-                case SyntaxKind.Minus:
-                    return "-";
+                case SyntaxKind.NumericLiteral:
+                    Debug.Fail("This should be handled by using the C# lexer's interned string in NumericLiteral()");
+                    return base.GetTokenContent(type);
                 case SyntaxKind.Not:
-                    return "!";
-                case SyntaxKind.Modulo:
-                    return "%";
-                case SyntaxKind.And:
-                    return "&";
                 case SyntaxKind.LeftParenthesis:
-                    return "(";
                 case SyntaxKind.RightParenthesis:
```

Comment:
Consider pulling the assert cases out into a separate local function to make the actual logic clearer.

Reply:
I'm honestly not sure what you mean here. How would that make the logic clearer?